### PR TITLE
Find a Supplier API - v1 - Add filters

### PR DESCRIPTION
**Pull request by jeannejx**
### Action Plan:

- [x] Get all filter items from the supplier collections to be shown in the front end.
- [x] Create language filter on user collection, return userId list. 
- [x] Create trait filter on listing collection, return userId list.
- [x] Create categories filter on listing collection return userId list.
- [x] Create attributes filter on supplier collection.
- [x] Create rating filter on supplier collection.
- [x]  Create name filter on supplier collection.
- [x]  Create keyword filter on bio field of supplier collection.
- [x] Create userId filter on the returning userId lists.
- [x] Should pass testing for the filter functions.

### CODE REVIEW

**Comment by codedavinci** [comment]
Jenny could you please have a more detailed breakdown of your Action Plan !

**Comment by jeannejx** [comment]
> Jenny could you please have a more detailed breakdown of your Action Plan !

sure working on it now.

File: src/models/supplier.ts [hunk 1]
```diff
@@ -133,4 +161,56 @@ export const SupplierSchema = new Schema<ISupplierSchema>(
   SchemaGlobalConfig
 )
 
-export default model<ISupplierSchema>('supplier', SupplierSchema)
+interface SupplierModel extends Model<ISupplierSchema> {
+  getAllSuppliers(): ISupplierSchema[]
```
Comment (codedavinci) [inline src/models/supplier.ts:165]: `getSupplier` and `getSuppliers`

File: src/models/__test__/listing.test.ts [hunk 2]
```diff
@@ -306,55 +306,79 @@ describe('Listing Model', () => {
...
+      addons: [
+        {
+          supplieID: new Types.ObjectId(),
+          rule: [{}]
+        }
+      ]
     })
-
-    expect(addon.rule[0].type).not.toBeDefined()
+    expect(listing.addons[0].rule[0].type).not.toBeDefined()
```
[the remark this answers is not in the document]
Comment (jeannejx) [reply src/models/__test__/listing.test.ts]: Has removed the code

File: src/models/supplier.ts [hunk 3]
```diff
@@ -113,6 +114,33 @@ export const SupplierSchema = new Schema<ISupplierSchema>(
     payoutMethods: { type: [PayoutMethodOptionSchema], default: () => [] },
     preferredPayoutMethod: { type: String, default: 'paypal' }, // will pick from the future preferred payout method collection
     rating: { type: RatingSchema, default: () => ({}) },
+    averageRating: {
+      type: Number,
+      default: function () {
+        if (
```
[the remark this answers is not in the document]
Comment (jeannejx) [reply src/models/supplier.ts]: Done

File: src/models/supplier.ts (hunk 1 above)
Comment (jeannejx) [reply src/models/supplier.ts]: Done

**Pull request closed without merging**